app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,404 @@
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+
+    const fieldSequence: FieldSequence = {
+        firstName: {
+            ref: firstNameRef,
+            isDisabled: lockedFirstName || includesSsoService(service),
+        },
+        lastName: {
+            ref: lastNameRef,
+            isDisabled: lockedLastName || includesSsoService(service),
+        },
+        username: {
+            ref: usernameRef,
+            isDisabled: service !== '',
+        },
+        email: {
+            ref: emailRef,
+            isDisabled: true,
+        },
+        nickname: {
+            ref: nicknameRef,
+            isDisabled: lockedNickname,
+        },
+        position: {
+            ref: positionRef,
+            isDisabled: lockedPosition,
+        },
+    };
+
+    const onFocusNextField = (fieldKey: string) => {
+        const findNextField = () => {
+            const fields = Object.keys(fieldSequence);
+            const curIndex = fields.indexOf(fieldKey);
+            const searchIndex = curIndex + 1;
+
+            if (curIndex === -1 || searchIndex > fields.length) {
+                return undefined;
+            }
+
+            const remainingFields = fields.slice(searchIndex);
+
+            const nextFieldIndex = remainingFields.findIndex((f: string) => {
+                const field = fieldSequence[f];
+                return !field.isDisabled;
+            });
+
+            if (nextFieldIndex === -1) {
+                Keyboard.dismiss();
+                return undefined;
+            }
+
+            const fieldName = remainingFields[nextFieldIndex];
+            return fieldSequence[fieldName];
+        };
+
+        if (fieldKey === 'position' && canSave) {
+            // performs form submission
+            Keyboard.dismiss();
+            submitUser();
+        } else {
+            const nextField = findNextField();
+            nextField?.ref?.current?.focus();
+        }
+    };
+
+    const fieldConfig: FieldConfig = {
+        blurOnSubmit: false,
+        enablesReturnKeyAutomatically: true,
+        onFocusNextField,
+        onTextChange: updateField,
+        returnKeyType: 'next',
+    };
+
+    if (updating) {
+        return <ProfileUpdating/>;
+    }
+
+    return (
+        <>
+            {isTablet &&
+                <TabletTitle
+                    action={buttonText}
+                    enabled={canSave}
+                    onPress={submitUser}
+                    testID='custom_status.done.button'
+                    title={intl.formatMessage({id: 'mobile.screen.your_profile', defaultMessage: 'Your Profile'})}
+                />
+            }
+            <SafeAreaView
+                edges={edges}
+                style={style.flex}
+                testID='edit_profile.screen'
+            >
+                <KeyboardAwareScrollView
+                    bounces={false}
```

Comment:
is this really working correctly on iOS/Tablet?

Reply:
Yes, but maybe I'm not looking at the same thing.  When I test it, it jumps from field to field.  Is there something that I should be looking more specifically at?

Reply:
Every focused input should be visible on screen. Ios android and tablets

Reply:
Please share a video of this working on an iPhone, an Android phone and on a tablet